Add in more tags to Fedora Cloud

jdoss opened the pull request:
plume: Adds in useful tags to Fedora Cloud and fixes Fedora Cloud download URL

This PR adds in two new tags to the Fedora Cloud AMI creation steps:

```
"Date":       specTimestamp,
￼"Arch":       specBoard,
```

and it fixes the download URL from https://kojipkgs.fedoraproject.org/compose/cloud/ 

This is the compose I tested my changes on below. https://koji.fedoraproject.org/koji/buildinfo?buildID=1361137

![image](https://user-images.githubusercontent.com/8195/63380186-93c34900-c35b-11e9-8a35-2ba0dec566a9.png)

```
[jdoss@sts71 mantle]$ ./bin/plume pre-release --system fedora --channel cloud --version 30 --timestamp 20190820 --respin 0 --board x86_64 --compose-id 1361137 --image-type Cloud-Base --debug
2019-08-20T19:32:14Z cli: Started logging at level DEBUG
2019-08-20T19:32:14Z plume: Running AWS pre-release...
2019-08-20T19:32:14Z plume: Reusing existing image "/home/jdoss/repos/opensource/mantle/.cache/images/cloud/30/Fedora-Cloud-Base-30-20190820.0.x86_64.raw"
2019-08-20T19:32:14Z plume: Connecting to AWS...
2019-08-20T19:32:15Z plume: Creating S3 object s3://ksec-my-cool-bucket/x86_64/30/Fedora-Cloud-Base-30-20190820.0.x86_64.raw...
2019-08-20T19:32:15Z platform/api/aws: skipping upload since object exists and force was not set: s3://ksec-my-cool-bucket/x86_64/30/Fedora-Cloud-Base-30-20190820.0.x86_64.raw
2019-08-20T19:32:15Z plume: Creating EBS snapshot...
2019-08-20T19:32:16Z platform/api/aws: created snapshot import task import-snap-0b3e6ecc605c8ef98
2019-08-20T19:32:16Z platform/api/aws: waiting for import task: active (2): pending
2019-08-20T19:32:36Z platform/api/aws: waiting for import task: active (2): pending
2019-08-20T19:32:56Z platform/api/aws: waiting for import task: active (2): pending
2019-08-20T19:33:16Z platform/api/aws: waiting for import task: active (2): pending
2019-08-20T19:33:36Z platform/api/aws: waiting for import task: active (5): validating
2019-08-20T19:33:56Z platform/api/aws: waiting for import task: active (18): validated
2019-08-20T19:34:16Z platform/api/aws: waiting for import task: active (32): validated
2019-08-20T19:34:36Z platform/api/aws: waiting for import task: active (32): validated
2019-08-20T19:34:57Z platform/api/aws: waiting for import task: active (35): downloading/converting
2019-08-20T19:35:17Z platform/api/aws: waiting for import task: active (35): downloading/converting
2019-08-20T19:35:37Z platform/api/aws: waiting for import task: active (35): downloading/converting
2019-08-20T19:35:57Z platform/api/aws: waiting for import task: active (37): downloading/converting
2019-08-20T19:36:17Z platform/api/aws: waiting for import task: active (40): downloading/converting
2019-08-20T19:36:37Z platform/api/aws: waiting for import task: active (40): downloading/converting
2019-08-20T19:36:58Z platform/api/aws: waiting for import task: active (40): downloading/converting
2019-08-20T19:37:18Z platform/api/aws: waiting for import task: active (40): downloading/converting
2019-08-20T19:37:38Z platform/api/aws: waiting for import task: active (43): downloading/converting
2019-08-20T19:37:58Z platform/api/aws: waiting for import task: active (70): converted
2019-08-20T19:38:18Z platform/api/aws: waiting for import task: active (72): converted
2019-08-20T19:38:38Z platform/api/aws: waiting for import task: active (75): preparing snapshot
2019-08-20T19:38:58Z platform/api/aws: waiting for import task: active (89): Preparing snapshot
2019-08-20T19:39:18Z platform/api/aws: waiting for import task: active (89): Preparing snapshot
2019-08-20T19:39:38Z platform/api/aws: waiting for import task: active (91): Preparing snapshot
2019-08-20T19:39:58Z platform/api/aws: waiting for import task: active (94): Preparing snapshot
2019-08-20T19:40:19Z platform/api/aws: waiting for import task: active (94): Preparing snapshot
2019-08-20T19:40:39Z platform/api/aws: waiting for import task: active (97): Preparing snapshot
2019-08-20T19:40:59Z plume: Deleting S3 object s3://ksec-my-cool-bucket/x86_64/30/Fedora-Cloud-Base-30-20190820.0.x86_64.raw...
2019-08-20T19:40:59Z platform/api/aws: deleting s3://ksec-my-cool-bucket/x86_64/30/Fedora-Cloud-Base-30-20190820.0.x86_64.raw
2019-08-20T19:41:00Z plume: Creating AMIs from snap-06c462ced7f54dffc...
2019-08-20T19:41:00Z plume: Replicating AMI ami-0903dee6a3ff99bc9...
2019-08-20T19:46:40Z plume: Running Azure pre-release...
2019-08-20T19:46:40Z plume: Azure image creation disabled.
2019-08-20T19:46:40Z plume: Pre-release complete, run `plume release` to finish.
```

arithx commented on cmd/plume/prerelease.go:
```diff
@@ -257,17 +257,36 @@ func getCLImageFile(client *http.Client, src *storage.Bucket, fileName string) (
 	}
 
-	rawxzURI, err := url.Parse(fmt.Sprintf("%v/%v/compose/%v/%v/images/%v", spec.BaseURL, specComposeID, getImageTypeURI(), specBoard, fileName))
+	rawxzURI, err := url.Parse(fmt.Sprintf("%v/%v/compose/%v/%v/images/%v", spec.BaseURL, imageBuildDir, getImageTypeURI(), specBoard, fileName))
```
Only the `user` & `cloud` channels have `spec.BuildDir` not empty. This probably needs to be gated to use the templated `imageBuildDir` on those channels while still using `specComposeID` on others.

jdoss replied:
Per our DM on IRC, I removed the Fedora Atomic Host channels and added `spec.BuildDir ` to the `branched` channel.

jdoss commented:
@arithx OK squashed with a better commit message.

dustymabe commented:
talked with @jdoss. we need some changes here. going to post a follow up or a new PR tomorrow or monday

jdoss commented:
OK I pushed the changes that @dustymabe and I talked about yesterday. We actually want to keep the rawhide, branched and updates channels. We also talked about removing the user channel that was added because we weren't sure it was used. We can add that back in later if needed. 

I also reverted the addition of `BuildDir` because I was using `--compose-id` totally wrong. I updated the README with a bunch of useful information to document what I have learned while working on this PR.

arithx commented:
At least in the context of CL the user channel is mostly for testing.

jdoss commented:
@dustymabe are we able to get this merged?

dustymabe commented:
ok finally got around to looking at this. Thanks for the patience @jdoss. 

I finally understand what the `user` channel is now. It was a way to use `awsFedoraUserPartitions` (i.e. our test account) instead of `awsFedoraPartitions` (our prod aws account). I kind of don't like the way it was being done in the past by using a channel. @arithx - is there any way to choose a different `awsPartitionSpec` without using a channel?

Either way I think we can probably merge this as is and do fixups in a follow up. Nice work @jdoss!

dustymabe commented on cmd/plume/README.md:
````diff
@@ -24,6 +139,25 @@ bin/plume pre-release -C user --verify-key ~/keyfile -V $version-$COREOS_BUILD_I
+Here is an example of doing a Fedora Cloud pre-release with plume:
+
+```
+./bin/plume pre-release \
````
do you have an example running the `release` step? I'm getting an error and I don't know if I'm doing something dumb.

dustymabe replied:
This is what I found to work (note that I had to use `--distro` and not `--system`: 

```
plume release --distro fedora --channel cloud --version 30 --timestamp 20190925 --respin 0 --board x86_64 --compose-id Fedora-Cloud-30-20190925.0 --image-type Cloud-Base --debug
```

Though that led to other errors.

jdoss replied:
What exact errors? I never did the release because I didn't want to make a bunch of AMIs public on my AWS account.

dustymabe replied:
i'm working through them now

dustymabe commented:
@arithx let's go ahead and merge this and I'll submit some followups
